### PR TITLE
tikv: add flow-control config override hint

### PR DESCRIPTION
**Pull request by Connor1996**
Signed-off-by: Connor1996 <zbk602423539@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add flow-control config override hint

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/tikv/tikv/pull/11840

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by tabokie — APPROVED** [review]

**Review by qiancai — APPROVED** [review]
LGTM

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -409,22 +409,22 @@ RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为
 
 ### `memtables-threshold`
 
-+ 当 KvDB 的 memtable 的个数达到该阈值时，流控机制开始工作。
++ 当 KvDB 的 memtable 的个数达到该阈值时，流控机制开始工作。当 `enable` 的值为 `true` 时，会覆盖 `rocksdb.(defaultcf|writecf|lockcf).max-write-buffer-number`。
```
Comment (en-jin19) [inline tikv-configuration-file.md:412]:
```suggestion
+ 当 KvDB 的 memtable 的个数达到该阈值时，流控机制开始工作。当 `enable` 的值为 `true` 时，`memtables-threshold` 的配置会覆盖 `rocksdb.(defaultcf|writecf|lockcf).max-write-buffer-number` 的配置。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -409,22 +409,22 @@ RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为
...
 + 默认值：5
 
 ### `l0-files-threshold`
 
-+ 当 KvDB 的 L0 文件个数达到该阈值时，流控机制开始工作。
++ 当 KvDB 的 L0 文件个数达到该阈值时，流控机制开始工作。当 `enable` 的值为 `true` 时，会覆盖 `rocksdb.(defaultcf|writecf|lockcf).level0-slowdown-writes-trigger`。
```
Comment (en-jin19) [inline tikv-configuration-file.md:417]:
```suggestion
+ 当 KvDB 的 L0 文件个数达到该阈值时，流控机制开始工作。当 `enable` 的值为 `true` 时，`l0-files-threshold` 的配置会覆盖 `rocksdb.(defaultcf|writecf|lockcf).level0-slowdown-writes-trigger` 的配置。
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -409,22 +409,22 @@ RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为
...
 + 默认值：20
 
 ### `soft-pending-compaction-bytes-limit`
 
-+ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制开始拒绝部分写入请求，报错 `ServerIsBusy`。
++ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制开始拒绝部分写入请求，报错 `ServerIsBusy`。当 `enable` 的值为 `true` 时，会覆盖 `rocksdb.(defaultcf|writecf|lockcf).soft-pending-compaction-bytes-limit`。
```
Comment (en-jin19) [inline tikv-configuration-file.md:422]:
```suggestion
+ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制开始拒绝部分写入请求，报错 `ServerIsBusy`。当 `enable` 的值为 `true` 时，`soft-pending-compaction-bytes-limit` 的配置会覆盖 `rocksdb.(defaultcf|writecf|lockcf).soft-pending-compaction-bytes-limit` 的配置。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -409,22 +409,22 @@ RocksDB 多个 CF 之间共享 block cache 的配置选项。当开启时，为
...
 + 默认值："192GB"
 
 ### `hard-pending-compaction-bytes-limit`
 
-+ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制拒绝所有写入请求，报错 `ServerIsBusy`。
++ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制拒绝所有写入请求，报错 `ServerIsBusy`。当 `enable` 的值为 `true` 时，会覆盖 `rocksdb.(defaultcf|writecf|lockcf).hard-pending-compaction-bytes-limit`。
```
Comment (en-jin19) [inline tikv-configuration-file.md:427]:
```suggestion
+ 当 KvDB 的 pending compaction bytes 达到该阈值时，流控机制拒绝所有写入请求，报错 `ServerIsBusy`。当 `enable` 的值为 `true` 时，`hard-pending-compaction-bytes-limit` 的配置会覆盖 `rocksdb.(defaultcf|writecf|lockcf).hard-pending-compaction-bytes-limit` 的配置。
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1143,7 +1143,7 @@ bloom filter 为每个 key 预留的长度。
 
 ### `max-write-buffer-number`
 
-+ 最大 memtable 个数。
++ 最大 memtable 个数。当 `storage.flow-control.enable` 的值为 `true` 时，`storage.flow-control.memtables-threshold` 会覆盖此配置。
```
Comment (en-jin19) [inline tikv-configuration-file.md:1146]:
```suggestion
+ 最大 memtable 个数。当 `storage.flow-control.enable` 的值为 `true` 时，`storage.flow-control.memtables-threshold` 会覆盖 `max-write-buffer-number` 的配置。
```

**Comment by en-jin19** [comment]
@Connor1996 PTAL, thanks!

**Review by en-jin19 — APPROVED** [review]
LGTM

**Pull request merged**